src/Shared/FileUtilities.cs

```diff
@@ -421,6 +421,72 @@ internal static string FixFilePath(string path)
         }
 
         /// <summary>
+        /// If on Unix, convert backslashes to slashes for strings that resemble paths.
+        /// The heuristic is if something resembles paths (contains slashes) check if the
+        /// first segment exists and is a directory.
+        /// Use a native shared method to massage file path. If the file is adjusted,
+        /// that qualifies is as a path.
+        /// </summary>
+        internal static string MaybeAdjustFilePath(string value)
+        {
+            if (NativeMethodsShared.IsWindows || string.IsNullOrWhiteSpace(value) ||
+                value.StartsWith("$(") || value.StartsWith("@(") || value.StartsWith("\\\\"))
+            {
+                return value;
```

Comment:
Since I change more than just fix slashes, I could only bypass the regex when there are no backslashes -- I would then have to check if the result has no forward slashes to skip the rest. Most strings that pass the first few tests are in fact paths, so that does not seem to gain much.